internal/scanner/scanner.go

```diff
@@ -89,27 +89,24 @@ func (s *Scanner) ScanNamespace(nsName string) error {
 	// create PolicyReport
 	namespacedsReport := report.NewPolicyReport(namespace)
 	namespacedsReport.Summary.Skip = skippedNum
+	// old policy report to be used as cache
+	previousNamespacedReport, err := s.reportStore.GetPolicyReport(nsName)
+	if err != nil {
+		log.Error().Err(err).Str("namespace", nsName).
+			Msg("error getting PolicyReport from store")
+	}
 
 	// Iterate through all auditableResources. Each item contains a list of resources and the policies that would need
 	// to evaluate them.
 	for i := range auditableResources {
-		auditResource(&auditableResources[i], &s.resourcesFetcher, &s.httpClient, &namespacedsReport)
-		err = s.reportStore.AddPolicyReport(&namespacedsReport)
+		auditResource(&auditableResources[i], s.resourcesFetcher, &s.httpClient, &namespacedsReport, &previousNamespacedReport)
+		err = s.reportStore.SavePolicyReport(&namespacedsReport)
 		if err != nil {
 			log.Error().Err(err).Msg("error adding PolicyReport to store")
 		}
 	}
 	log.Info().Str("namespace", nsName).Msg("scan finished")
-
-	if s.printJSON {
```

Comment:
if we aren't using printJSON anymore, then I would remove the cli flag, and the cmd tests too.

But as stated in the previous comment I would prefer to have both the in memory store, and a new client store, that both implement an interface for saving the reports.